### PR TITLE
[hipRTC] add lowest() for float to MIOpen custom limits

### PR DESCRIPTION
**Pull request by junliume**
#2752

### CODE REVIEW

File: src/kernels/miopen_limits.hpp [hunk 1]
```diff
@@ -46,6 +46,8 @@ class numeric_limits<float>
     static constexpr __device__ float max() noexcept { return 0x1.FFFFFEp+127f; }
 
     static constexpr __device__ float min() noexcept { return 0x1p-126f; }
+
+    static constexpr __device__ float lowest() noexcept { return -0x1.fffffep+127; }
```
Comment (junliume) [inline src/kernels/miopen_limits.hpp:51]: This line *might* need to be protected by `HIP_PACKAGE_VERSION_FLAT` but let's see if it works as is.

Comment (junliume) [reply src/kernels/miopen_limits.hpp:51]: Added anyway since the earliest trace of issue comes together with the one below on int

**Review by JehandadKhan — APPROVED** [review]

File: src/kernels/miopen_limits.hpp (hunk 1 above)
Comment (atamazov) [reply src/kernels/miopen_limits.hpp:51]: @junliume It is highly unlikely that in some HIP release (or release candidate) we have PCH with `lowest()`, while `min()` and `max()` are missing there. Therefore I recommend removing the conditional wrapper (`#if HIP_PACKAGE_VERSION_FLAT >= 6001024024ULL` and `#endif`).

**Comment by amberhassaan** [comment]
@junliume , @atamazov : slightly off topic but what do you guys think of using constants from `<cfloat>` (https://en.cppreference.com/w/cpp/header/cfloat) which is what is used to define `numeric_limists` (see further down the page (https://en.cppreference.com/w/cpp/types/numeric_limits). This way we don't have to maintain constants ourselves?

**Comment by atamazov** [comment]
@amberhassaan (Maybe I've misunderstood your idea, but) hipRTC does not allow any standard headers to be used in compiled programs, including `<cfloat>`. See https://reviews.llvm.org/D100652 for more info.

**Comment by amberhassaan** [comment]
> @amberhassaan (Maybe I've misunderstood your idea, but) hipRTC does not allow any standard headers to be used in compiled programs, including `<cfloat>`. See https://reviews.llvm.org/D100652 for more info.

No, you're right. I did think of it after typing my comment. I was hoping that we can slip in some simple C header that just defines some constants but technically we can't rely on that. The only other option would be to embed these system headers with MIOpen the way we do for user-defined include files.

**Pull request merged**